### ISSUE TITLE
GroupBy async throws System.ArgumentNullException: Value cannot be null.

### ISSUE THREAD

**Issue by sam-wheat** [aspnet/EntityFrameworkCore#11561]
I need to de-dupe a very large table.  I am using the query below which throws the exception shown below.
Project attached.
```

	using (Db db = new Db(dbContextOptions))
	{

		var query = db.Sources
			.GroupBy(x => new { x.DataProviderID, x.NativeID })
			.Select(x => x.First()) //remove dupes
			.GroupBy(x => x.DataProviderID);


		await query.ForEachAsync(async grp =>  // throws
		{
			await Task.FromResult(0);
		});

	}


System.ArgumentNullException: Value cannot be null.
Parameter name: parameter
   at System.Linq.Expressions.Expression.Lambda(Expression body, String name, Boolean tailCall, IEnumerable`1 parameters)
   at Microsoft.EntityFrameworkCore.Query.ResultOperatorHandler.HandleGroup(EntityQueryModelVisitor entityQueryModelVisitor, GroupResultOperator groupResultOperator, QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.ResultOperatorHandler.HandleResultOperator(EntityQueryModelVisitor entityQueryModelVisitor, ResultOperatorBase resultOperator, QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.Internal.RelationalResultOperatorHandler.HandleGroup(HandlerContext handlerContext)
   at Microsoft.EntityFrameworkCore.Query.Internal.RelationalResultOperatorHandler.HandleResultOperator(EntityQueryModelVisitor entityQueryModelVisitor, ResultOperatorBase resultOperator, QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.VisitResultOperator(ResultOperatorBase resultOperator, QueryModel queryModel, Int32 index)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor.VisitResultOperator(ResultOperatorBase resultOperator, QueryModel queryModel, Int32 index)
   at Remotion.Linq.QueryModelVisitorBase.VisitResultOperators(ObservableCollection`1 resultOperators, QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.VisitQueryModel(QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.RelationalQueryModelVisitor.VisitQueryModel(QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.EntityQueryModelVisitor.CreateAsyncQueryExecutor[TResult](QueryModel queryModel)
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.ExecuteAsync[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryable`1.System.Collections.Generic.IAsyncEnumerable<TResult>.GetEnumerator()
   at System.Linq.AsyncEnumerable.<ForEachAsync_>d__174`1.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at InsertTest.Program.<CreateRelease2>d__5.MoveNext() in C:\Users\sam\Documents\Projects\GroupByTest\InsertTest\Program.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at InsertTest.Program.<Run>d__4.MoveNext() in 

C:\Users\sam\Documents\Projects\GroupByTest\InsertTest\Program.cs:line 56
```

[GroupByTest.zip](https://github.com/aspnet/EntityFrameworkCore/files/1881753/GroupByTest.zip)

**Comment by smitpatel** [comment]
```C#
    public partial class Source
    {
        public int ID { get; set; }
        public int NativeID { get; set; }
        public int DataProviderID { get; set; }
        public int UserID { get; set; }
        public string Name { get; set; }
        public string Link { get; set; }
        public string Notes { get; set; }
    }
```
Query
```
var query = db.Sources
    .GroupBy(x => new { x.DataProviderID, x.NativeID })
    .Select(x => x.First())
    .GroupBy(x => x.DataProviderID)
    .ToListAsync().GetAwaiter().GetResult();
```
Sync version passes. Async version fails. Probably cause: null CancellationToken parameter.

**Comment by sam-wheat** [comment]
@smitpatel  Thank you for the fast response!  Will we see this fix in 2.1?

**Comment by smitpatel** [comment]
Yes, it will be in 2.1 release.
